extensions/windows-event-log/wel/WindowsEventLog.cpp

```diff
@@ -25,64 +25,67 @@
 #include "UnicodeConversion.h"
 #include "utils/Deleters.h"
 #include "utils/gsl.h"
+#include "UniqueEvtHandle.h"
 
 namespace org::apache::nifi::minifi::wel {
 
+namespace {
+std::string GetEventTimestampStr(uint64_t event_timestamp) {
+  SYSTEMTIME st;
+  FILETIME ft;
+
+  ft.dwHighDateTime = (DWORD)((event_timestamp >> 32) & 0xFF'FF'FF'FF);
+  ft.dwLowDateTime = (DWORD)(event_timestamp & 0xFF'FF'FF'FF);
+
+  FileTimeToSystemTime(&ft, &st);
+  std::stringstream datestr;
+
+  std::string period = "AM";
+  auto hour = st.wHour;
+  if (hour >= 12 && hour < 24)
+    period = "PM";
+  if (hour > 12)
+    hour -= 12;
+  if (hour == 0)
+    hour = 12;
+  datestr << st.wMonth << "/" << st.wDay << "/" << st.wYear << " " << std::setfill('0') << std::setw(2) << hour << ":" << std::setfill('0')
+          << std::setw(2) << st.wMinute << ":" << std::setfill('0') << std::setw(2) << st.wSecond << " " << period;
+  return datestr.str();
+}
+}  // namespace
```

Comment:
Is there any way to convert the timestamp using standard date/time formatting facilities, or date.h? Because these are complex problems with existing solutions, and I can't confidently say that this covers all edge cases. (endianness, timezones, DST and so on)

Reply:
fortunetly there is :), originally I didnt want to touch this in this PR (only extracted the function), but it turns out its quite simple to make it much more readable, and easier to maintain with date.h
So I've refactored this aswell in https://github.com/apache/nifi-minifi-cpp/pull/1434/commits/e142a7d9796ac63ccd4758a9e5a570dae1509fc9#diff-d148f5f0fe3bb6a877f5754ff0567450d155c4e21d65e28292966f40bb0b77a4R34-R40

Reply:
This is awesome, thanks!